### PR TITLE
Replace query params of GET persons/ with 'term'

### PR DESCRIPTION
**Pull request by sdiv877**
Addresses #161 and #181:
* Replaces the query param scheme of GET persons/ with a single 'term' param
* Changed response if a user's `encounters`/`persons` field is empty, to an empty array with (`200 OK`)

Manual Postman Tests:
* Empty user `encounters`/`persons` fields return empty arrays (`200 OK`) ✔️ 
* Only persons that belong to the `User` found through the `auth_id` are returned (`200 OK`) ✔️ 
* If a `User` is not found returns `404 Not Found` ✔️ 
* If a `User` has no `persons` returns `204 No Content` ✔️ 
* Partial matches of the provided `term` query are applied ✔️ 
    * `first_name`
    * `last_name`
    * `gender`
    * `location`
    * `how_we_met`
    * `organisation`
 * If  the `term` param is empty it is ignored ✔️

To be fixed later:
* Add real unit tests to replace manual tests

### CODE REVIEW

File: backend/src/services/person.service.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import mongoose from 'mongoose';
 import Person, { PersonModel } from '../models/person.model';
 import logger from '../utils/logger';
 
-const stringFields = ['full_name', 'gender', 'location', 'how_we_met', 'organisation'];
```
Comment (lin8231) [inline backend/src/services/person.service.ts:8]: How come gender is not a field anymore?

Comment (sdiv877) [reply backend/src/services/person.service.ts:8]: Whoops should be fixed now

**Review by lanxdrey19 — DISMISSED** [review]
LGTM

**Review by lanxdrey19 — APPROVED** [review]
LGTM

**Review by jimwang6012 — APPROVED** [review]
LGTM!

**Pull request merged**